contracts/Strategy.sol

```diff
@@ -159,56 +144,112 @@ contract Strategy is BaseStrategy {
         warningLTVMultiplier = _warningLTVMultiplier;
         acceptableCostsRay = _acceptableCostsRay;
         maxTotalBorrowIT = _maxTotalBorrowIT;
-        if (_aaveReferral != 0) {
-            referral = _aaveReferral;
-        }
+        referral = _aaveReferral;
         _setIsInvestmentTokenIncentivised(_isInvestmentTokenIncentivised);
         _setIsWantIncentivised(_isWantIncentivised);
         leaveDebtBehind = _leaveDebtBehind;
         require(maxLoss <= 10_000);
         maxLoss = _maxLoss;
     }
 
-    function setYVault(IVault _yVault, uint256 _maxLoss)
-        external
-        onlyGovernance
-    {
-        if (balanceOfDebt() != 0) {
-            yVault.withdraw(
-                yVault.balanceOf(address(this)),
-                address(this),
-                _maxLoss
-            ); // we withdraw the full amount from investmentToken vault
-            _repayInvestmentTokenDebt(balanceOfInvestmentToken()); // we use all of our balance to repay debt with Aave
-
-            // we sell profits
-            if (balanceOfInvestmentToken() > 0) {
-                _sellInvestmentForWant(balanceOfInvestmentToken());
-            }
+    event Cloned(address indexed clone);
+
+    function clone(
```

Comment:
use cloneAaveLender() or something like that.
let's be specific with the clone method so it's not confused with the BaseStrategy one.